validphys2/src/validphys/n3fit_data.py

```diff
@@ -102,23 +102,68 @@ def replica_luxseed(replica, luxseed):
     return replica_nnseed(replica, luxseed)
 
 
-class _TrMasks(TupleComp):
+class _Masks(TupleComp):
     """Class holding the training validation mask for a group of datasets
     If the same group of dataset receives the same trvlseed then the mask
     will be the same.
     This class holds said information so it can be reused easily, i.e.,
     ``group_name`` and ``seed`` define the ``masks``.
+
+    In case of the diagonal basis, the ``rotation`` is a tuple of the eigenvalues and
+    eigenvectors of the fitting covariance matrix.
     """
 
-    def __init__(self, group_name, seed, masks=None):
-        self.masks = masks
+    def __init__(
+        self, group_name, seed, tr_masks, vl_masks, diagonal_basis=False, eig_vals=None, u=None
+    ):
+
+        self.tr_masks = tr_masks
+        self.vl_masks = vl_masks
+        if diagonal_basis:
+            self.eig_vals = eig_vals
+            self.u = u
+
         super().__init__(group_name, seed)
 
-    def __iter__(self):
-        yield from self.masks
 
+def _diagonal_masks(
+    data,
+    replica_trvlseed,
+    dataset_inputs_fitting_covmat,
+    diagonal_basis=False,
```

Comment:
Does diagonal basis have an effect here? (at the end of the call it is set to true) I guess this is a remnant of the previous version?